### PR TITLE
ActionList Fixes

### PR DESCRIPTION
**Pull request by dgreif**
Addressing a number of issues found when integrating `ActionList` into `SelectPanel`

- [x] Divider should be hidden above a group that has a filled header
- [x] `inline` description is not inline
- [x] Multi-select action lists should use checkboxes
  - [x] Checkbox instead of checkmark icon
  - [x] Properly align checkbox vertically
  - [x] ~Wrap item in a `<label>`?~ (Using `aria-label={text}` to avoid branching DOM and for succinct announcements)
  - [x] Prevent tabbing to checkbox if in focusZone (this one might be tough)
  - [x] ~Update docs with new `selectionVariant` option~ (Prop is commented in the `ItemProps` interface definition, but `ActionList.Item` does not have a corresponding `.mdx` file) 

### Screenshots
| Before | After |
|--------|--------|
| image | image |
| image | image |  
| image | image |

### Merge checklist
- [ ] Added/updated tests
- [ ] Added/updated documentation
- [x] Tested in Chrome
- [ ] Tested in Firefox
- [x] Tested in Safari
- [ ] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/components/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

**Review by VanAnderson — APPROVED** [review]
LGTM!

File: .changeset/blue-items-repeat.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+"@primer/components": patch
+---
+
+Hide divider before `ActionList.Group`s with filled header
+Align `Item` description to when rendered in-line
+Add `selectionVariant: 'multiple'` for `Item`s.  These will use a checkbox input instead of a checkmark icon for selected state
+Allow `focusZoneSettings` to be passed into `AnchoredOverlay`
```
Comment (colebemis) [inline .changeset/blue-items-repeat.md:8]: If it's not too much trouble, it might be nice to put all these distinct changes in their own changeset.

File: src/ActionList/List.tsx [hunk 2]
```diff
@@ -82,6 +87,12 @@ export type ListProps = ListPropsBase | GroupedListProps
 
 const StyledList = styled.div`
   font-size: ${get('fontSizes.1')};
+  /* 14px font-size * 1.428571429 = 20px line height
+   *
+   * TODO: When rem-based spacing on a 4px scale lands, replace
+   * hardcoded '1.428571429'
+   */
+  line-height: 1.428571429;
```
Comment (colebemis) [inline src/ActionList/List.tsx:95]: Could we just hard code 20px here instead?
```suggestion
  line-height: 20px;
```

**Review by colebemis — APPROVED** [review]
Looks great, @dgreif! Thanks for cleaning this up ❤️ 

Just left a few non-blocking comments.

File: src/ActionList/List.tsx (hunk 2 above)
Comment (dgreif) [reply src/ActionList/List.tsx:95]: @smockle thoughts on this?

Comment (smockle) [reply src/ActionList/List.tsx:95]: I’m fine either way: The hardcoded value is going to be replaced with a variable after the [“new scaling and typography internal variables”](https://github.com/github/primer/issues/41#issuecomment-839264461) land. cc @vdepizzol for visibility

Comment (dgreif) [reply src/ActionList/List.tsx]: Thanks @smockle.  I made the switch to `20px` since it's a little easier to read for now.
